### ISSUE TITLE
Async Continuation pattern fails in certain scenarios (ex, Asp.Net pipeline)

### ISSUE THREAD

**Issue by angelatan2** [newrelic/newrelic-dotnet-agent#189]
**Description**
Many of our async instrumentation wrappers rely on applying a continuation to the Task object returned from the instrumented method.

The continuation processes the actual result from the method call and records information about the timing, success, etc of the segment and can close the transaction.

In some cases, this continuation may not fire.  For example, it seems that the ASP.NET pipeline sometimes does not process the continuations.  We think it is because it is killing off the execution context/scheduler based on the completion of the request.

This results in transactions not being reported/transformed.

Consider the following code example that is part of a WebAPI application.   The GetAsyncFireAndForget sporadically does not fire the continuation.  The GetTaskRunFireAndForget seems to work well.

```
public class ValuesController : ApiController
	{
		[HttpGet]
		[Route("api/values/GetAsyncFireAndForget")]
		public string GetAsyncFireAndForget()
		{
			BackgroundWork();

			SomeOtherMethod();
			return "ok";
		}

		[HttpGet]
		[Route("api/values/GetTaskRunFireAndForget")]
		public string GetTaskRunFireAndForget()
		{
			Task.Run(() => BackgroundWork());
			SomeOtherMethod();
			return "ok";
		}

		[Trace]
		private void SomeOtherMethod()
		{
			Thread.Sleep(5000);
		}

		[Trace]
		private void SomeOtherMethod2()
		{
			Thread.Sleep(5000);
		}

		[Transaction]
		private async Task BackgroundWork()
		{
			await Task.Delay(5000);
			NewRelic.Api.Agent.NewRelic.AddCustomParameter("key", "value");
			SomeOtherMethod2();
		}
	}
```

Suggestion:
Modify async wrappers in the following ways
Handle situations where the async return is not of type Task, Task<T>.   Specifically void.  Cannot just return.
Identify whether or not to support use cases that are not Task.Run, Task.Factory, or new Thread()

**Comment by bgrainger** [comment]
> In some cases, this continuation may not fire.

This may be addressed by the changes I suggested here https://github.com/newrelic/newrelic-dotnet-agent/issues/85#issuecomment-689952690 to schedule the continuation on the default `TaskScheduler` with `TaskContinuationOptions.ExecuteSynchronously`.

**Comment by angelatan2** [comment]
We need to try and reproduce and verify that the changes from #85 may have solved this issue.

**Comment by tehbio** [comment]
I've done some testing on the current build with the provided code as well as some other async patterns and have not encountered this described problem. 

Seems any inconsistency with following a continuation in one of these scenarios has been resolved by a related change included in PR #85, so I'm closing this issue.
